engine/apps/alerts/escalation_snapshot/snapshot_classes/escalation_policy_snapshot.py

```diff
@@ -105,6 +105,13 @@ def next_user_in_sorted_queue(self) -> User:
         next_user = users_queue[(last_user_index + 1) % len(users_queue)]
         return next_user
 
+    def has_finished_log_record(self, alert_group_id: int) -> bool:
+        return AlertGroupLogRecord.objects.filter(
+            escalation_policy_id=self.id,
+            alert_group_id=alert_group_id,
+            type=AlertGroupLogRecord.TYPE_ESCALATION_FINISHED,
```

Comment:
This type of logs doesn't have `escalation_policy_id` because we create it when there are no more escalation steps left

Reply:
good catch 👍 this should be addressed in [this commit](https://github.com/grafana/oncall/pull/1266/commits/c78e45f48f3a1f0099f969a7cce8b69b020b78e5)